zwave_js_server/firmware.py

```diff
@@ -0,0 +1,35 @@
+"""Firmware update helper."""
+from typing import Any
+
+import aiohttp
+
+from .client import Client
+from .model.node import Node
+from .util.helpers import convert_bytes_to_base64
+
+
+async def begin_firmware_update(
+    url: str,
+    node: Node,
+    filename: str,
+    file: bytes,
+    session: aiohttp.ClientSession,
+    file_format: str = None,
+) -> Any:
+    """Send beginFirmwareUpdate command to Node."""
+    client = Client(url, session)
+    await client.connect()
+    await client.set_api_schema()
+
+    cmd = {
+        "command": "node.begin_firmware_update",
+        "nodeId": node.node_id,
+        "firmwareFilename": filename,
+        "firmwareFile": convert_bytes_to_base64(file),
+    }
+    if file_format is not None:
+        cmd["firmwareFileFormat"] = file_format
+
+    resp = await client.async_send_command(cmd, require_schema=5)
+    await client.disconnect()
+    return resp
```

Comment:
What is the response? How will it be used?

Reply:
If everything is successful, the response is an empty dictionary. There are cases where one of the underlying methods will fail, like if the file format can't be guessed, the firmware can't be extracted, or the command to begin the update fails. In those cases a `ZwaveError` should be thrown which we should catch and return as the response

Reply:
Here's an example of what was returned when I sent thee server an invalid payload for thee file:
```
{
  type: 'result',
  success: false,
  messageId: 'update-firmware',
  errorCode: 'zwave_error',
  zwaveErrorCode: 1506,
  zwaveErrorMessage: 'Could not detect firmware format'
}
```

Reply:
We should decide what type of return value we want and ehat errors to raise. I don't think we should return the response directly.

Reply:
I don't think this is specific to this feature. Maybe we should have a generic exception that is raised when a `zwave_error` is received?

Reply:
This is how we do it in the client.

https://github.com/home-assistant-libs/zwave-js-server-python/blob/aaa18a6fdb942de5aa2e9670d7789ef725242342/zwave_js_server/client.py#L269-L271

Reply:
Do we need to do anything then? I think the failure scenario I imagined is already handled

Reply:
Yes, right, it's already handled. If we don't have any use for the response on non error, we don't need to return anything I think.